storage/scylla_api.c

```diff
@@ -0,0 +1,302 @@
+
+void print_error(CassFuture* future) {
+  const char* message;
+  size_t message_length;
+  cass_future_error_message(future, &message, &message_length);
+  fprintf(stderr, "Error: %.*s\n", (int)message_length, message);
+}
+
+CassCluster* create_cluster(const char* hosts) {
+  CassCluster* cluster = cass_cluster_new();
+  cass_cluster_set_contact_points(cluster, hosts);
+  return cluster;
+}
+
+CassError connect_session(CassSession* session, const CassCluster* cluster) {
+  CassError rc = CASS_OK;
+  CassFuture* future = cass_session_connect(session, cluster);
+
+  cass_future_wait(future);
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  }
+  cass_future_free(future);
+
+  return rc;
+}
+
+CassError execute_query(CassSession* session, const char* query) {
+  CassError rc = CASS_OK;
+  CassStatement* statement = cass_statement_new(query, 0);
+  CassFuture* future = cass_session_execute(session, statement);
+
+  cass_future_wait(future);
+
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  }
+
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return rc;
+}
+
+CassError prepare_query(CassSession* session, const char* query, const CassPrepared** prepared) {
+  CassError rc = CASS_OK;
+  CassFuture* future = NULL;
+
+  future = cass_session_prepare(session, query);
+  cass_future_wait(future);
+
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  } else {
+    *prepared = cass_future_get_prepared(future);
+  }
+
+  cass_future_free(future);
+
+  return rc;
+}
+
+CassError insert_into_bundleTable(CassSession* session, const CassPrepared* prepared, const Transaction* transaction) {
+  CassError rc = CASS_OK;
+  CassStatement* statement = NULL;
+  CassFuture* future = NULL;
+
+  statement = cass_prepared_bind(prepared);
+  cass_statement_bind_bytes_by_name(statement, "bundle", transaction->bundle, strlen((char*)transaction->bundle));
+  cass_statement_bind_bytes_by_name(statement, "address", transaction->address, strlen((char*)transaction->address));
+  cass_statement_bind_bytes_by_name(statement, "transaction", transaction->transaction,
+                                    strlen((char*)transaction->transaction));
+  cass_statement_bind_string_by_name(statement, "message", transaction->message);
+  cass_statement_bind_int64_by_name(statement, "value", transaction->value);
+  cass_statement_bind_bytes_by_name(statement, "timestamp", transaction->timestamp,
+                                    strlen((char*)transaction->timestamp));
+
+  future = cass_session_execute(session, statement);
+
+  cass_future_wait(future);
+
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  }
+
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return rc;
+}
+static int mallocAndCopyBlob(cass_byte_t** target, const cass_byte_t* src, size_t size) {
```

Comment:
Maybe make the return value of this function to `void`?
The return value is not used when calling it.

Reply:
I think I could modify it to check whether malloc failure?